City-of-Helsinki/drupal-helfi-kasvatus-koulutus
# UHF-5002 Insert high school group menu to mobile menu

# Insert high school group menu to mobile menu [UHF-5002](https://helsinkisolutionoffice.atlassian.net/browse/UHF-5002)
High schools have their own group menus that are visible on the sidebar when viewing the page on desktop. They were not visible on mobile menu so we needed to inject them there with javascript.

## What was done
* Using javascript the group menu is injected under a link item in the main main that has the class "high-school-parent".

## How to install
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* Checkout this branch.
* Run `make drush-cr`

## How to test
* First of all - I recommend testing this feature with the latest database from **testing**. There is already two groups done so the testing is so much easier.
* Log into the site and go to edit the main menu: https://helfi-kasko.docker.so/fi/kasvatus-ja-koulutus/admin/structure/menu/manage/main
* Find a link that says "Lukiot" - this is going to be the menu item that the group menu will be injected under. Edit the link and add it class under options: "high-school-parent".
* Now go to see the groups: https://helfi-kasko.docker.so/fi/kasvatus-ja-koulutus/admin/group. If you are using the testing database there should be two. You can pick either one. 
* On the group page select the tab "Nodes". From there select a basic page and go to view it.
* If the page has sidebar menu on it then it will be good for our test: Resize the browser window until you get the mobile menu. 
* Open the mobile menu and you should see the Lukiot-link active and open with the high school and its sublinks added under it. They should work just like any other part of the menu works.
